### ISSUE TITLE
Event macros with no fields shouldn't require `{}`

### ISSUE THREAD

**Issue by hawkw** [hawkw/tokio-trace-prototype#74]
Follow-up from PR #62.

Currently, when creating an event with no fields, an empty `{}` is required. For example,
```rust
event!(Level::Debug, {}, "i have no fields!");
```
It would be nice if the `{}` could be elided in this case, and the macro invoked as
```rust
event!(Level::Debug, "i have no fields!");
``` 
This is especially relevant in light of #62, which adds `trace!`, `debug!`, `info!`, `warn!`, and `error!` macros similar to those in the `log` crate --- if those macros no longer require the curly braces, then creating an event with no fields becomes identical to invoking a `log` macro. This would make it very easy to begin transitioning a crate using `log` to use `tokio-trace`. 

A library author would probably want to go back and edit these macro invocations to pass fields in a structured way after replacing `#[macro_use] extern crate log` with `#[macro_use] extern crate tokio-trace`, but it would mean that the crate would compile after making that one-line change. It would also allow a crate to define a feature flag which switches between `log` and `tokio-trace` without having to introduce a layer of indirection, which is nice for libraries which want to support `tokio-trace` without breaking compatibility.

This _should_ be possible with the current way macros match their inputs, although I think it may require matching a `$fmt:expr` token prior to `$($arg:tt)+` to disambiguate the matches.

**Comment by hawkw** [comment]
@davidbarsky is this something you're interested in looking into?

**Comment by davidbarsky** [comment]
@hawkw yeah, I can work on this.

**Comment by hawkw** [comment]
@davidbarsky cool, it's not in core, so this isn't something that needs to be done right away.

**Comment by davidbarsky** [comment]
Closed by #62.
